WooCommerce/Classes/ViewRelated/Coupons/Add and Edit Coupons/AddEditCouponViewModel.swift

```diff
@@ -0,0 +1,93 @@
+import Foundation
+import Yosemite
+
+/// View model for `AddEditCoupon` view
+///
+final class AddEditCouponViewModel: ObservableObject {
+
+    private let siteID: Int64
+    private let editingOption: EditingOption
+    private let discountType: Coupon.DiscountType
+    var titleView: String {
+        switch editingOption {
+        case .creation:
+            switch discountType {
+            case .percent:
+                return Localization.titleCreatePercentageDiscount
+            case .fixedCart:
+                return Localization.titleCreateFixedCardDiscount
+            case .fixedProduct:
+                return Localization.titleCreateFixedProductDiscount
+            default:
+                return Localization.titleCreateGenericDiscount
+            }
+        case .editing:
+            switch discountType {
+            case .percent:
+                return Localization.titleEditPercentageDiscount
+            case .fixedCart:
+                return Localization.titleEditFixedCardDiscount
+            case .fixedProduct:
+                return Localization.titleEditFixedProductDiscount
+            default:
+                return Localization.titleEditGenericDiscount
+            }
+        }
```

Comment:
We have an [extension](https://github.com/woocommerce/woocommerce-ios/blob/daba32d494d3f8dcba6c84d3c4c55c515edd80ab/WooCommerce/Classes/Model/Coupon%2BWoo.swift#L4) for Coupon Discount Type where the localized name for each type is defined. How about moving this to that extension to keep the view model lightweight?

Reply:
Another thing: We will not enable editing and creating coupons with unsupported discount types - so maybe there's no need for generic content.